util/conditions/patch.go

```diff
@@ -97,16 +97,6 @@ func (o *applyOptions) isOwnedCondition(t clusterv1.ConditionType) bool {
 // ApplyOption defines an option for applying a condition patch.
 type ApplyOption func(*applyOptions)
 
-// WithOwnedConditions allows to define condition types owned by the controller.
-// In case of conflicts for the owned conditions, the patch helper will always use the value provided by the controller.
-//
-// DEPRECATED: Use WithForceOverwrite.
-func WithOwnedConditions(t ...clusterv1.ConditionType) ApplyOption {
```

Comment:
I need to double-check this, because after some back and forth, I think that in the end, we deemed it necessary for KCP/MHC to update machine conditions (it should not force override all the conditions in this cases)